server/services/version.ts

```diff
@@ -86,13 +87,22 @@ interface CreateVersion {
   metadata: any
 }
 
-export async function createVersion(user: UserDoc, data: CreateVersion) {
+export async function createVersion(user: UserDoc, data: CreateVersion, parentUuid: string) {
```

Comment:
Services shouldn't interact with other domains.  E.g. here, the `Version` service interacts with the `Model` domain.  Instead, you should use the `Model` service (or expect the calling process to pass in a full Model instance).